### PR TITLE
Remove dependency on enzyme from components/App

### PR DESCRIPTION
**Pull request by suhail34**
## Which problem is this PR solving?
- Part of #1668
- Using React Testing Library to remove dependency on enzyme for index.test.js and TopNav.test.js

## How was this change tested?
- Tested my changes inside packages/jaeger-ui/ using command `npx jest --testPathPattern=src/components/App/.*\.test\.js`
- Also performed yarn test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]

File: packages/jaeger-ui/src/components/App/TopNav.test.js [hunk 1]
```diff
@@ -55,85 +56,110 @@ describe('<TopNav>', () => {
...
       });
 
       it('renders sub-menu links', () => {
         dropdownItems.slice(1, 2).forEach(itemConfig => {
-          const item = subMenu.dive().find(`[href="${itemConfig.url}"]`);
-          expect(item.length).toBe(1);
-          expect(item.prop('target')).toBe(itemConfig.anchorTarget || '_blank');
-          expect(item.text()).toBe(itemConfig.label);
+          //fireEvent.mouseEnter(screen.getByText(labelAbout))
+          console.log(itemConfig.label);
```
Comment (anshgoyalevil) [inline packages/jaeger-ui/src/components/App/TopNav.test.js:158]:
```suggestion
```

Comment (anshgoyalevil) [inline packages/jaeger-ui/src/components/App/TopNav.test.js:158]:
```suggestion
```

**Comment by yurishkuro** [comment]
@suhail34 are you planning to finish this PR?

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**